### ISSUE TITLE
isValid() doesn't recognize abbreviated French month strings for format "MMM" when locale == "fr".

### ISSUE THREAD

**Issue by hrimhari** [moment/moment#3126]
moment("Jan", "MMM", "en", true).isValid() will return true as expected.

After setting moment.locale("fr"), moment().format( "MMM" ) returns ["jan.", "fév.", ...] instead of ["Jan",  "Feb", ...].

moment("Jan", "MMM", "fr", true).isValid() and moment("jan", "MMM", true).isValid() returns false, which is expected.

But moment("jan.", "MMM", "fr", true).isValid() will also return false. "jan." is the same string used by format(), so why was it not accepted as valid?

There doesn't seem to be a way to validate abbreviated month for "fr" locale.

From Developer Console:

> moment.locale("en");
> <- "en"
> moment(Date.parse("2016-01-05")).format("MMM");
> <- "Jan"
> moment("Jan", "MMM", "en", true).isValid();
> <- true
> moment.locale("fr");
> <- "fr"
> moment(Date.parse("2016-01-05")).format("MMM");
> <- **"janv."**
> moment("Jan", "MMM", "fr", true).isValid();
> <- false
> moment(**"janv."**, "MMM", "fr", true).isValid();
> <- false

**Comment by pkyeck** [comment]
same for me ... switching from 2.12 to 2.13 changed behavior of this code snippet:

```javascript
const moment = require("moment");
moment.locale("de");
moment.updateLocale("de", {
  monthsShort : "Jan._Feb._Mrz._Apr._Mai_Jun._Jul._Aug._Sep._Okt._Nov._Dez.".split("_")
});

let date = moment("01. Mrz 18", ["DD. MMM YY"], true);
console.log(date.isValid());

let date1 = moment("01. Mrz. 18", ["DD. MMM YY"], true);
console.log(date1.isValid());
```

output 2.12
```javascript
true
false
false
```

output 2.13
```javascript
false
true
false
```

so in my eyes this should have been a major update :(

**Comment by maggiepint** [comment]
So, just getting around to this one now.

@hrimhari, when I run that code, I see the following:

```js
moment.locale('fr')
"fr"
moment("Jan", "MMM", "fr", true).isValid();
false
moment("janv.", "MMM", "fr", true).isValid();
true
```

This is what I would expect, given that the french short month abbreviation we have on record is in fact 'janv.'.

On that same note, I would expect the following to be invalid:

```js
moment("jan.", "MMM", "fr", true).isValid()
```

You are parsing in french, and the short month for French is 'janv.' not 'jan.'.

Am I missing something here?

**Comment by maggiepint** [comment]
@pkyeck I'm not sure why you have three result sets. I only see two. In any case, if I run your code (including your update locale statement) I see the following:

```js
moment("01. Mrz 18", ["DD. MMM YY"], true).isValid()
false
moment("01. Mrz. 18", ["DD. MMM YY"], true).isValid()
true
```

This is again what I would expect since there is a ``.`` at the end of Mrz that we expect to match.

This IS a change from previous versions, in the sense that parsing any month abbreviation with a ``.`` was broken before and now it's not.

If you want the following behavior: 
```js
moment("01. Mrz 18", ["DD. MMM YY"], true).isValid()
true
moment("01. Mrz. 18", ["DD. MMM YY"], true).isValid()
false
```

You could run the following code:

```js
moment.updateLocale("de", {
  monthsShort : "Jan._Feb._Mrz._Apr._Mai_Jun._Jul._Aug._Sep._Okt._Nov._Dez.".split("_"), 
monthsParseExact:false
});
```

That monthsParseExact param is what @ichernev added to make the strict parsing work. To be honest, I'm not entirely sure of what the implications of toggling it back are, but I *think* it should result in the old behavior.

**Comment by pkyeck** [comment]
@maggiepint thanks for the reply.
the third output was something else I had in my test script and I just forgot to remove the last `false` from the output.

> This IS a change from previous versions, in the sense that parsing any month abbreviation with a . was broken before and now it's not.

Good that we agree on that ;)
But my point is, that it was a major change, that breaks existing code when you have `^2.12.x` in your `package.json`. If you follow semver this should have been released as `3.0.0` to not break older code snippets - even if it was broken before. People were relying on this broken behavior.

I just downgraded back to 2.12 and pinned the version in my `package.json`.
